scipy/stats/_binned_statistic.py

```diff
@@ -598,13 +602,25 @@ def binned_statistic_dd(sample, values, statistic='mean',
             result[vv, a] = flatsum
     elif statistic == 'median':
         result.fill(np.nan)
-        _calc_binned_statistic(Vdim, binnumbers, result, values, np.median)
+        if values[0][0] != None:
+            from ._calc_binned_statistic_pythran import _calc_binned_statistic_pythran
+            _calc_binned_statistic_pythran(Vdim, binnumbers, result, values, statistic)
+        else:
+            _calc_binned_statistic(Vdim, binnumbers, result, values, np.median)
     elif statistic == 'min':
         result.fill(np.nan)
-        _calc_binned_statistic(Vdim, binnumbers, result, values, np.min)
+        if values[0][0] != None:
+            from ._calc_binned_statistic_pythran import _calc_binned_statistic_pythran
+            _calc_binned_statistic_pythran(Vdim, binnumbers, result, values, statistic)
+        else:
+            _calc_binned_statistic(Vdim, binnumbers, result, values, np.min)
```

Comment:
I think this whole is-else block needs to live within `def _calc_binned_statistic` in this file`. And regarding names:

- `_calc_binned_statistic_pythran` is the main function, right?
- `_calc_binned_statistic` needs to handle: (a) custom callables, and (b) `object` dtype. (anything else?)

It would be great to add a comment explaining that.

Reply:
Yes! Current`_calc_binned_statistic_pythran`  basically has the same logic as `_calc_binned_statistic` but with an additional check for `statistics`. 

In my first two versions(https://github.com/scipy/scipy/pull/14345/commits/503a30659a86e9cc87467421af7e5b933d2c8dad, https://github.com/scipy/scipy/pull/14345/commits/6926cd7081730676b1d58f2c39189215c8eb22e8), `_calc_binned_statistic_pythran` was to handle the whole `if-else` block(including `mean,count,sum`), but the codes became much more ugly after adding checks to fix the `object` type problem. So I changed it to the current version :)